lib/internal/util/inspect.js

```diff
@@ -1542,6 +1542,54 @@ function reduceToSingleString(
   return `${braces[0]}${ln}${join(output, `,\n${indentation}  `)} ${braces[1]}`;
 }
 
+function getToStringValue(value, inspectOptions) {
+  const yes = 2;
+  const maybe = 1;
+  const no = 0;
+  let isBuiltIn = yes;
+  // Count objects that have no `toString` function as built-in.
+  if (typeof value.toString === 'function') {
+    let pointer = value;
+    isBuiltIn = no;
+    // Check if the object has a direct own `toString` property or if the
+    // prototype that contains the `toString` property or a former prototype is
+    // a built-in object or not.
+    while (!hasOwnProperty(pointer, 'toString')) {
+      const descriptor = Object
+        .getOwnPropertyDescriptor(pointer, 'constructor');
+      if (descriptor !== undefined &&
+          typeof descriptor.value === 'function' &&
+          builtInObjects.has(descriptor.value.name)) {
+        isBuiltIn = yes;
+        break;
+      }
```

Comment:
I'm not sure but is this part needed? Wouldn't the part after the loop check for the same thing but on the latest prototype?

Reply:
It does have two very small differences:

1. The inspection would ends as soon as any built-in is detected, not when the `toString` is detected (different performance profile - some parts in the prototype chain do not have an own `toString` function).
2. A built-in that has no own `toString` function with a manipulated prototype (non built-in one that contains a own `toString` function) would be handled differently.

I originally implemented it like this for the 1. reason. In this particular case there is no absolute right or wrong, since it'll be overhead in some cases while in other cases it will be less work and the other way around for the other implementation.

I am fine to remove this part to keep the code more concise.

Reply:
Okay then, I'd prefer more concise implementation (considering that built-in's prototype manipulation is usually not a good idea and IMO standalone classes should usually dominate the extended built-in ones) but I'm totally fine with both variants, just wanted to make sure this is intentional.

Also if we go with the current variant I think your points above should go in the comment above this `if` to make sure people wouldn't misunderstand later on.

Reply:
Sure, I refactored the code in general for simplicity. PTAL